telink: tl721x: fix tercel pm mode.

haiwentelink opened the pull request:
- change power mode to DCDC_0P94_DCDC_1P8.
- set frequency to 80Mhz.
- set core to 0.9v ,and low down retention ldo.

andriy-bilynskyy commented:
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| hal_telink | https://github.com/telink-semi/hal_telink/commit/d45ad1d5772528d8e3efc23fdcf8478cab7956f9 | https://github.com/telink-semi/hal_telink/commit/8e116bb74c344c043500425793d76c4d3ae21565 (develop) | [telink-semi/hal_telink@d45ad1d5..8e116bb7](https://github.com/telink-semi/hal_telink/compare/d45ad1d5772528d8e3efc23fdcf8478cab7956f9..8e116bb74c344c043500425793d76c4d3ae21565) |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

damien0x0023 commented on soc/telink/tlsr/telink_tlx/soc.c:
```diff
@@ -162,12 +167,28 @@ void soc_early_init_hook(void)
+
+#if CONFIG_SOC_RISCV_TELINK_TL721X
+	case CLK_120MHZ:
+		PLL_240M_CCLK_120M_HCLK_60M_PCLK_60M_MSPI_60M;
```
PLL_240M_CCLK_120M_HCLK_60M_PCLK_60M_MSPI_48M

haiwentelink replied:
fixed

damien0x0023 commented on soc/telink/tlsr/telink_tlx/soc.c:
```diff
@@ -162,12 +167,28 @@ void soc_early_init_hook(void)
+		PLL_240M_CCLK_120M_HCLK_60M_PCLK_60M_MSPI_60M;
+		break;
+	case CLK_240MHZ:
+		PLL_240M_CCLK_240M_HCLK_120M_PCLK_120M_MSPI_60M;
```
PLL_240M_CCLK_240M_HCLK_120M_PCLK_120M_MSPI_48M

haiwentelink replied:
fixed

damien0x0023 commented on soc/telink/tlsr/telink_tlx/soc.c:
```diff
@@ -224,12 +245,25 @@ void soc_tlx_restore(void)
+
+#if CONFIG_SOC_RISCV_TELINK_TL721X
+	case CLK_120MHZ:
+		PLL_240M_CCLK_120M_HCLK_60M_PCLK_60M_MSPI_60M;
```
PLL_240M_CCLK_120M_HCLK_60M_PCLK_60M_MSPI_48M

haiwentelink replied:
fixed

damien0x0023 commented on soc/telink/tlsr/telink_tlx/soc.c:
```diff
@@ -224,12 +245,25 @@ void soc_tlx_restore(void)
+		PLL_240M_CCLK_120M_HCLK_60M_PCLK_60M_MSPI_60M;
+		break;
+	case CLK_240MHZ:
+		PLL_240M_CCLK_240M_HCLK_120M_PCLK_120M_MSPI_60M;
```
PLL_240M_CCLK_240M_HCLK_120M_PCLK_120M_MSPI_48M

haiwentelink replied:
fixed

damien0x0023 commented on soc/telink/tlsr/telink_tlx/soc.c:
```diff
@@ -32,6 +32,7 @@
 #define CLK_60MHZ  60000000u
 #define CLK_80MHZ  80000000u
 #define CLK_120MHZ 120000000u
+#define CLK_240MHZ 240000000u
```
"(DT_PROP(DT_PATH(cpus, cpu_0), clock_frequency) != CLK_240MHZ" is missing in Line84

haiwentelink replied:
fixed

damien0x0023 reviewed (commented):
We need to revert MSPI frequency in this PR, too.

s07641069 commented on soc/telink/tlsr/telink_tlx/soc.c:
```diff
@@ -162,12 +167,28 @@ void soc_early_init_hook(void)
+		PLL_240M_CCLK_240M_HCLK_120M_PCLK_120M_MSPI_60M;
+		break;
+#endif
+
```
fix this compliance check warning
looks like empty line is left

haiwentelink replied:
already delete

s07641069 commented on soc/telink/tlsr/telink_tlx/soc.c:
```diff
@@ -162,12 +167,28 @@ void soc_early_init_hook(void)
 		break;
 #endif /* CONFIG_SOC_RISCV_TELINK_TL321X */
+
+
```
same here 
fix this compliance check warning
looks like empty line is left

haiwentelink replied:
already delete

s07641069 commented on soc/telink/tlsr/telink_tlx/soc.c:
```diff
@@ -80,8 +81,9 @@
 	 (DT_PROP(DT_PATH(cpus, cpu_0), clock_frequency) != CLK_80MHZ) && \
-	 (DT_PROP(DT_PATH(cpus, cpu_0), clock_frequency) != CLK_120MHZ))
-#error "Invalid clock-frequency. Supported values: 24, 40, 48, 60, 80 and 120 MHz"
+	 (DT_PROP(DT_PATH(cpus, cpu_0), clock_frequency) != CLK_120MHZ) && \
```
here is new compliance warning
you can check hidden (optional) compliance warnings under modified files page: 
https://github.com/telink-semi/zephyr/pull/476/files

haiwentelink replied:
@s07641069  I try some ways , currently just follow the sytle as B9X did , for the tips in ci , i think it can not fit , I also try just copy the suggestion as the ci , it will also fail .

haiwentelink committed:
hal: telink: add support for 60M Mspi

- add clock setting for 60M mspi.
- select 60M mspi can save power.
- adjust power table for buteo and tercel.

Signed-off-by: Haiwen Xia <haiwen.xia@telink-semi.com>
